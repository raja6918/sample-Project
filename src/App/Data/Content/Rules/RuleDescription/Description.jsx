import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getComponent, getValidation } from './constants';
import Notification from '../../../../../components/Notification';
import * as ruleService from '../../../../../services/Data/rules';
import ReactMarkdown from 'react-markdown';

const HyperLink = styled.span`
  color: #0a75c2;
  text-decoration: underline;
  cursor: pointer;
`;

export class Description extends PureComponent {
  state = {
    error: false,
    snackMessage: null,
    errorField: '',
  };

  componentWillReceiveProps(nextProps) {
    /**
     * To remove error state when overlay in outer component is removed
     */
    if (nextProps.overlay !== this.state.error && this.state.error) {
      this.setState({ error: nextProps.overlay, snackMessage: null });
    }
  }

  isTooltipDisabled = (value, data) =>
    !data.definedInCurrent || this.props.readOnly;

  getTooltipContent = (value, data) =>
    !data.definedInCurrent
      ? this.props.t('DATA.rules.tooltip.settingsFrom', {
          name: data.referenceRulesetName,
        })
      : this.props.t('DATA.rules.tooltip.revertTo', {
          name: data.referenceRulesetName,
        });

  handleDisable = (value, data) => this.props.readOnly;

  handleOnchange = async (value, data) => {
    try {
      // Here we can write logic for custum local validation based on input type and name
      const validation = getValidation(data);
      if (validation) validation(this.props.t, value, data);

      // this.props.toggleLoader(true);
      // Call setPram API
      await ruleService.setParam(
        this.props.userId,
        this.props.scenarioId,
        this.props.ruleSet,
        data.name,
        value
      );
      // Once API call is success we update the global state
      this.props.handleParamSet(value, data, this.props.code);

      if (this.state.error) {
        this.setState({ error: false });
      }
    } catch (error) {
      if (error.response) {
        this.setState({ error: true, errorField: data.name }, () => {
          this.props.reportError(error);
          this.props.toggleLoader(false);
          this.props.setOverlay();
        });
      } else {
        // For handling client validation side error
        this.setState(
          { error: true, errorField: data.name, snackMessage: error.message },
          () => this.props.setOverlay()
        );
      }

      console.error(error);
    }
  };

  onClearSnackBar = () => {
    this.setState({ snackMessage: null });
  };

  /**
   * To remove parent component overlay when someone click on rendered generic component
   */
  removeOverlay = () => {
    if (this.props.overlay) {
      this.props.removeOverlay();
      this.onClearSnackBar();
    }
  };

  handleParamReset = data => {
    this.props.handleParamReset(data, this.props.ruleSet, this.props.code);
  };

  customRenderers = {
    paragraph: 'span',
    link: props => {
      const parsedData = this.props.data.references[
        props.children[0].props.value
      ];
      const component = parsedData ? getComponent(parsedData) : null;
      if (component) {
        const CustomComponent = component.component;

        return (
          <Fragment>
            <CustomComponent
              {...component.props}
              error={
                this.state.error && this.state.errorField === parsedData.name
              }
              data={parsedData}
              onChange={this.handleOnchange}
              handleDisable={this.handleDisable}
              handleReset={this.handleParamReset}
              handleTooltipDisable={this.isTooltipDisabled}
              getTooltipContent={this.getTooltipContent}
              removeOverlay={this.removeOverlay}
              scenarioId={this.props.scenarioId}
              userId={this.props.userId}
              ruleSet={this.props.ruleSet}
              reportError={this.props.reportError}
              refreshRules={this.props.refreshRules}
            />
            <Notification
              message={this.state.snackMessage}
              clear={this.onClearSnackBar}
              type={'error'}
              autoHideDuration={360000}
            />
          </Fragment>
        );
      } else if (parsedData && parsedData.type === 'hyperlink') {
        return (
          <HyperLink
            onClick={() =>
              this.props.handleLink(parsedData, this.props.ruleSet)
            }
          >
            {parsedData.value}
          </HyperLink>
        );
      } else {
        return <Fragment />;
      }
    },
  };

  render() {
    const { data } = this.props;

    if (data) {
      return (
        <ReactMarkdown renderers={this.customRenderers} source={data.data} />
      );
    }
    return <Fragment />;
  }
}

Description.propTypes = {
  t: PropTypes.func.isRequired,
  code: PropTypes.string.isRequired,
  data: PropTypes.shape().isRequired,
  ruleSet: PropTypes.number.isRequired,
  handleParamSet: PropTypes.func.isRequired,
  handleParamReset: PropTypes.func.isRequired,
  handleLink: PropTypes.func.isRequired,
  overlay: PropTypes.bool.isRequired,
  setOverlay: PropTypes.func.isRequired,
  removeOverlay: PropTypes.func.isRequired,
  userId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  scenarioId: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  reportError: PropTypes.func.isRequired,
  readOnly: PropTypes.bool.isRequired,
  toggleLoader: PropTypes.func.isRequired,
  refreshRules: PropTypes.func.isRequired,
};

export default Description;
