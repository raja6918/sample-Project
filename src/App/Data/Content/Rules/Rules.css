.MuiDrawer-paper {
  overflow-y: hidden;
}
.newRulecls .MuiFormLabel-root {
  font-size: 0.8rem;
}
.newRulecls label + .MuiInput-formControl {
  margin-top: 0;
}
.newRulecls .MuiInputLabel-formControl {
  transform: none;
}
.drawerCls .MuiPaper-root {
  background-color: #efefef;
}
.more-option .MuiPaper-elevation8 {
  box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2) !important;
}
.titleCls {
  top: 6px;
  position: absolute;
  left: -8px;
  font-size: 1.1rem;
  max-width: 235px;
}
.subTitleCls {
  position: absolute;
  left: -8px;
  font-size: 1.2em;
  font-weight: 400;
  line-height: 1.2;
}
.rootSubTitleCls {
  position: absolute;
  left: 39px;
  font-size: 1.2em;
  font-weight: 400;
}
.basicTtlcls {
  font-size: 1.1rem;
  position: absolute;
  left: 47px;
  top: 12px;
  max-width: 235px;
}
.basicImg {
  left: 14px;
  width: 21px;
  position: absolute;
  top: 24px;
}
.displayImg {
  display: none;
}

.ruleSetClass {
  width: 219px;
  left: 10px;
}
.ruleSetClass label {
  font-size: 0.9rem;
}
.container-cls {
  padding-bottom: 5px;
}
.ruleset-select {
  min-width: 217px;
}
.tree-icon {
  z-index: 999;
}
.ruleset-tooltip {
  overflow: hidden;
  text-overflow: ellipsis;
  display: block !important;
}
