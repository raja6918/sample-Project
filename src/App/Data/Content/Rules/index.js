import RulesWithErrorHandling from './Rules';

export default RulesWithErrorHandling;
