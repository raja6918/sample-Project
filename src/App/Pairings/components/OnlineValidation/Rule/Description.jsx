import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import ReactMarkdown from 'react-markdown';
import { connect } from 'react-redux';
import Notification from '../../../../../components/Notification';
import {
  getComponent,
  getValidation,
} from '../../../../Data/Content/Rules/RuleDescription/constants';
import * as ruleService from '../../../../../services/Data/rules';
import { checkPermission } from '../../../../../utils/common';
import scopes from '../../../../../constants/scopes';

export class Description extends PureComponent {
  state = {
    error: false,
    errorField: '',
    snackMessage: null,
  };

  componentWillReceiveProps(nextProps) {
    /**
     * To remove error state when overlay in outer component is removed
     */
    if (nextProps.overlay !== this.state.error && this.state.error) {
      this.setState({ error: nextProps.overlay, snackMessage: null });
    }
  }

  isTooltipDisabled = (value, data) =>
    !data.definedInCurrent || this.props.readOnly;

  getTooltipContent = (value, data) =>
    !data.definedInCurrent
      ? this.props.t('DATA.rules.tooltip.settingsFrom', {
          name: data.referenceRulesetName,
        })
      : this.props.t('DATA.rules.tooltip.revertTo', {
          name: data.referenceRulesetName,
        });

  handleDisable = (value, data) => {
    const { readOnly, permissions } = this.props;

    return readOnly || !checkPermission(scopes.rules.ruleEdit, permissions);
  };

  handleOnchange = async (value, data) => {
    try {
      // Here we can write logic for custum local validation based on input type and name
      const validation = getValidation(data);
      if (validation) validation(this.props.t, value, data);

      this.props.toggleLoader(true);

      // Call setPram API
      await ruleService.setParam(
        this.props.userId,
        this.props.scenarioId,
        this.props.ruleSet,
        data.name,
        value
      );
      // Once API call is success we update the global state
      this.props.handleParamSet(data.name, value, this.props.ruleSet);

      if (this.state.error) {
        this.setState({ error: false, errorField: '' });
      }
    } catch (error) {
      if (error.response) {
        this.setState({ error: true, errorField: data.name }, () => {
          this.props.reportError({ error });
          this.props.setOverlay();
        });
      } else {
        // For handling client validation side error
        this.setState(
          { error: true, errorField: data.name, snackMessage: error.message },
          () => this.props.setOverlay()
        );
      }

      console.error(error);
    } finally {
      this.props.toggleLoader(false);
    }
  };

  onClearSnackBar = () => {
    this.setState({ snackMessage: null });
  };

  /**
   * To remove parent component overlay when someone click on rendered generic component
   */
  removeOverlay = () => {
    if (this.props.overlay) {
      this.props.removeOverlay();
      this.onClearSnackBar();
    }
  };

  handleParamReset = data => {
    this.props.handleParamReset(data, this.props.ruleSet, this.props.code);
  };

  customRenderers = {
    paragraph: 'span',
    link: props => {
      const parsedData = this.props.data.references[
        props.children[0].props.value
      ];
      const component = getComponent(parsedData);
      if (component) {
        const CustomComponent = component.component;

        return (
          <Fragment>
            <CustomComponent
              {...component.props}
              error={
                this.state.error && this.state.errorField === parsedData.name
              }
              data={parsedData}
              onChange={this.handleOnchange}
              handleDisable={this.handleDisable}
              handleReset={this.handleParamReset}
              handleTooltipDisable={this.isTooltipDisabled}
              getTooltipContent={this.getTooltipContent}
              removeOverlay={this.removeOverlay}
            />
            <Notification
              message={this.state.snackMessage}
              clear={this.onClearSnackBar}
              type={'error'}
              autoHideDuration={360000}
            />
          </Fragment>
        );
      }

      if (parsedData.type === 'hyperlink') {
        return <Fragment>{parsedData.value}</Fragment>;
      }

      if (parsedData.type === 'text') {
        return <Fragment>{parsedData.value}</Fragment>;
      }
    },
  };

  render() {
    const { data } = this.props;

    if (data) {
      return (
        <ReactMarkdown renderers={this.customRenderers} source={data.data} />
      );
    }
    return <Fragment />;
  }
}

const mapStateToProps = state => {
  return { permissions: state.user.permissions };
};

Description.propTypes = {
  t: PropTypes.func.isRequired,
  code: PropTypes.string.isRequired,
  data: PropTypes.shape().isRequired,
  ruleSet: PropTypes.number.isRequired,
  handleParamSet: PropTypes.func.isRequired,
  handleParamReset: PropTypes.func.isRequired,
  overlay: PropTypes.bool.isRequired,
  setOverlay: PropTypes.func.isRequired,
  removeOverlay: PropTypes.func.isRequired,
  userId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  scenarioId: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  readOnly: PropTypes.bool.isRequired,
  toggleLoader: PropTypes.func.isRequired,
  reportError: PropTypes.func.isRequired,
  permissions: PropTypes.shape([]).isRequired,
};

export default connect(mapStateToProps)(Description);
