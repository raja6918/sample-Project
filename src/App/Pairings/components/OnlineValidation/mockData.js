export const alerts = {
  74: {
    alertLevel: 'error',
    alerts: {
      errorAlerts: [
        {
          rule: null,
          message: 'Invalid flight reference',
          tag: 'at',
          leg: {
            flightNumber: 8992,
            departureStationCode: 'YHZ',
            arrivalStationCode: 'YYT',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
        {
          rule: null,
          message: 'Mismatched arrival and departure stations between flights.',
          tag: 'at',
          leg: {
            flightNumber: 8899,
            departureStationCode: 'YYR',
            arrivalStationCode: 'YHZ',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
      warningAlerts: [
        {
          rule: { code: 'duty-credit', name: 'Duty credit' },
          message: 'duty credit exceeded',
          tag: 'at',
          leg: {
            flightNumber: 8829,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
      cautionAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 8885,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
      infoAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 8634,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
    },
  },
  115: {
    alertLevel: 'warning',
    alerts: {
      warningAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Maximum legs per duty' },
          message: 'maximum legs per duty exceeded',
          tag: 'at',
          leg: {
            flightNumber: 7951,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
        {
          rule: { code: 'duty-credit', name: 'Duty credit' },
          message: 'duty credit exceeded',
          tag: 'at',
          leg: {
            flightNumber: 7954,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
    },
  },
  116: {
    alertLevel: 'caution',
    alerts: {
      cautionAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 7954,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
    },
  },
  120: {
    alertLevel: 'info',
    alerts: {
      infoAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 7951,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
    },
  },
  176: {
    alertLevel: 'caution',
    alerts: {
      cautionAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 7957,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
      infoAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 7965,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
    },
  },
};
