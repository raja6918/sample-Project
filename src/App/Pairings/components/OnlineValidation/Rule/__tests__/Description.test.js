import React from 'react';
import { shallow } from 'enzyme';
import { Description } from '../Description';
import { ruleDescriptionData } from '../../../../../Data/Content/Rules/mockData';
import * as ruleService from '../../../../../../services/Data/rules';

describe('Test Pairing Alerts Rule Description Component', () => {
  const minProps = {
    t: jest.fn(msg => msg),
    handleParamSet: jest.fn(),
    handleParamReset: jest.fn(),
    setOverlay: jest.fn(),
    removeOverlay: jest.fn(),
    reportError: jest.fn(),
    toggleLoader: jest.fn(),
    handleRemoveParamSet: jest.fn(),
  };

  test('Description Component renders text correctly', () => {
    const wrapper = shallow(
      <Description
        code={ruleDescriptionData[0].code}
        data={ruleDescriptionData[0].userDescription[2]}
        {...minProps}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Description Component renders HyperLink correctly', () => {
    const data = ruleDescriptionData[0].userDescription[1];
    const wrapper = shallow(
      <Description
        code={ruleDescriptionData[0].code}
        data={data}
        ruleSet={ruleDescriptionData[0].ruleset}
        {...minProps}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test.skip('GenericTextField Component render and work correctly', () => {
    const code = ruleDescriptionData[0].code;
    const data = ruleDescriptionData[0].userDescription[0];
    const wrapper = shallow(
      <Description
        code={code}
        data={data}
        ruleSet={ruleDescriptionData[0].ruleset}
        overlay={true}
        {...minProps}
      />
    );
    expect(wrapper).toMatchSnapshot();

    // Check whether component call handleOnchange when onChange is triggered
    wrapper.instance().handleOnchange = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('GenericTextField').prop('onChange')(data.value, data);
    expect(wrapper.instance().handleOnchange).toBeCalledWith(data.value, data);

    // Check whether component call handleParamReset props when handleReset is triggered
    wrapper.find('GenericTextField').prop('handleReset')(data);
    expect(minProps.handleParamReset).toBeCalledWith(
      data,
      ruleDescriptionData[0].ruleset,
      code
    );

    // Check whether component call removeOverlay props when removeOverlay is triggered
    wrapper.find('GenericTextField').prop('removeOverlay')();
    expect(minProps.removeOverlay).toBeCalledWith();
  });

  test.skip('Check whether snackMessage set to null when onClearSnackBar method triggered', async () => {
    const code = ruleDescriptionData[0].code;
    const data = ruleDescriptionData[0].userDescription[0];
    const wrapper = shallow(
      <Description
        code={code}
        data={data}
        ruleSet={ruleDescriptionData[0].ruleset}
        overlay={false}
        {...minProps}
      />
    );

    wrapper.setState({ snackMessage: 'Test message' });
    wrapper.find('Notification').prop('clear')();
    expect(wrapper.state('snackMessage')).toBe(null);
  });

  test.skip('GenericTimeField Component render and work correctly', async () => {
    const code = ruleDescriptionData[1].code;
    const data = ruleDescriptionData[1].userDescription[1];
    const wrapper = shallow(
      <Description
        code={code}
        data={data}
        ruleSet={ruleDescriptionData[0].ruleset}
        overlay={false}
        {...minProps}
      />
    );
    expect(wrapper).toMatchSnapshot();

    // Check whether component call handleOnchange when onChange is triggered
    wrapper.instance().handleOnchange = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('GenericTimeField').prop('onChange')(data.value, data);
    expect(wrapper.instance().handleOnchange).toBeCalledWith(data.value, data);

    // Check whether component call handleParamReset props when handleReset is triggered
    wrapper.find('GenericTimeField').prop('handleReset')(data);
    expect(minProps.handleParamReset).toBeCalledWith(
      data,
      ruleDescriptionData[0].ruleset,
      code
    );

    // Check whether component call removeOverlay props when removeOverlay is triggered
    wrapper.find('GenericTimeField').prop('removeOverlay')();
    expect(minProps.removeOverlay).toBeCalledWith();
  });

  test.skip('GenericDateField Component render and work correctly', async () => {
    const code = ruleDescriptionData[7].code;
    const data = ruleDescriptionData[7].userDescription[1];
    const wrapper = shallow(
      <Description
        code={code}
        data={data}
        ruleSet={ruleDescriptionData[0].ruleset}
        overlay={true}
        {...minProps}
      />
    );
    expect(wrapper).toMatchSnapshot();

    // Check whether component call handleOnchange when onChange is triggered
    wrapper.instance().handleOnchange = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('GenericDateField').prop('onChange')(data.value, data);
    expect(wrapper.instance().handleOnchange).toBeCalledWith(data.value, data);

    // Check whether component call handleParamReset props when handleReset is triggered
    wrapper.find('GenericDateField').prop('handleReset')(data);
    expect(minProps.handleParamReset).toBeCalledWith(
      data,
      ruleDescriptionData[0].ruleset,
      code
    );

    // Check whether component call removeOverlay props when removeOverlay is triggered
    wrapper.find('GenericDateField').prop('removeOverlay')();
    expect(minProps.removeOverlay).toBeCalledWith();
  });

  test.skip('SelectInput Component render and work correctly', async () => {
    const code = ruleDescriptionData[3].code;
    const data = ruleDescriptionData[3].userDescription[1];
    const wrapper = shallow(
      <Description
        code={code}
        data={data}
        ruleSet={ruleDescriptionData[0].ruleset}
        overlay={true}
        {...minProps}
      />
    );
    expect(wrapper).toMatchSnapshot();

    // Check whether component call handleOnchange when onChange is triggered
    wrapper.instance().handleOnchange = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('SelectInput').prop('onChange')(data.value, data);
    expect(wrapper.instance().handleOnchange).toBeCalledWith(data.value, data);

    // Check whether component call handleParamReset props when handleReset is triggered
    wrapper.find('SelectInput').prop('handleReset')(data);
    expect(minProps.handleParamReset).toBeCalledWith(
      data,
      ruleDescriptionData[0].ruleset,
      code
    );

    // Check whether component call removeOverlay props when removeOverlay is triggered
    wrapper.find('SelectInput').prop('removeOverlay')();
    expect(minProps.removeOverlay).toBeCalledWith();
  });

  test.skip('check whether local validation is triggered when invalid data is passed in handleOnchange', async () => {
    const data = ruleDescriptionData[1].userDescription[1];
    const wrapper = shallow(
      <Description
        t={msg => msg}
        code={ruleDescriptionData[0].code}
        data={ruleDescriptionData[1].userDescription[1]}
        overlay={false}
        {...minProps}
      />
    );

    await wrapper.instance().handleOnchange('05:00', data);

    expect(wrapper.state('error')).toBe(true);
    expect(wrapper.state('snackMessage')).toBe(
      'ERRORS.RULES_DESCRIPTION.Time.rangeValidation'
    );
  });

  test.skip('check whether reportError is called when error occured in setParam Service', async () => {
    const data = ruleDescriptionData[3].userDescription[1];
    const wrapper = shallow(
      <Description
        code={ruleDescriptionData[0].code}
        data={ruleDescriptionData[0].userDescription[2]}
        overlay={false}
        {...minProps}
      />
    );

    ruleService.setParam = jest.fn(() =>
      // eslint-disable-next-line prefer-promise-reject-errors
      Promise.reject({
        response: {},
      })
    );

    await wrapper.instance().handleOnchange(data.value, data);

    expect(minProps.reportError).toHaveBeenCalled();
  });
});
