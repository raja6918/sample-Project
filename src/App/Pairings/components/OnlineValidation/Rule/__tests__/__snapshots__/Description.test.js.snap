// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Pairing Alerts Rule Description Component Description Component renders HyperLink correctly 1`] = `<Fragment />`;

exports[`Test Pairing Alerts Rule Description Component Description Component renders text correctly 1`] = `<Fragment />`;

exports[`Test Pairing Alerts Rule Description Component GenericDateField Component render and work correctly 1`] = `
<ReactMarkdown
  astPlugins={Array []}
  escapeHtml={true}
  parserOptions={Object {}}
  plugins={Array []}
  rawSourcePos={false}
  renderers={
    Object {
      "link": [Function],
      "paragraph": "span",
    }
  }
  skipHtml={false}
  sourcePos={false}
  transformLinkUri={[Function]}
/>
`;

exports[`Test Pairing Alerts Rule Description Component GenericTextField Component render and work correctly 1`] = `
<ReactMarkdown
  astPlugins={Array []}
  escapeHtml={true}
  parserOptions={Object {}}
  plugins={Array []}
  rawSourcePos={false}
  renderers={
    Object {
      "link": [Function],
      "paragraph": "span",
    }
  }
  skipHtml={false}
  sourcePos={false}
  transformLinkUri={[Function]}
/>
`;

exports[`Test Pairing Alerts Rule Description Component GenericTimeField Component render and work correctly 1`] = `
<ReactMarkdown
  astPlugins={Array []}
  escapeHtml={true}
  parserOptions={Object {}}
  plugins={Array []}
  rawSourcePos={false}
  renderers={
    Object {
      "link": [Function],
      "paragraph": "span",
    }
  }
  skipHtml={false}
  sourcePos={false}
  transformLinkUri={[Function]}
/>
`;

exports[`Test Pairing Alerts Rule Description Component GenericTimeField Component render and work correctly 2`] = `
<ReactMarkdown
  astPlugins={Array []}
  escapeHtml={true}
  parserOptions={Object {}}
  plugins={Array []}
  rawSourcePos={false}
  renderers={
    Object {
      "link": [Function],
      "paragraph": "span",
    }
  }
  skipHtml={false}
  sourcePos={false}
  transformLinkUri={[Function]}
/>
`;

exports[`Test Pairing Alerts Rule Description Component SelectInput Component render and work correctly 1`] = `
<ReactMarkdown
  astPlugins={Array []}
  escapeHtml={true}
  parserOptions={Object {}}
  plugins={Array []}
  rawSourcePos={false}
  renderers={
    Object {
      "link": [Function],
      "paragraph": "span",
    }
  }
  skipHtml={false}
  sourcePos={false}
  transformLinkUri={[Function]}
/>
`;
